[BCHDCC-40]: Focus Order & [BCHDCC-41]: Link Purpose

emmabjj opened the pull request:
## Description
<!--- Describe your changes in some detail by answering questions such as: -->
<!---   Why is this change needed? -->
<!---   How does it address the issue?-->

WCAG 2.4.3: As a user, I want to be able to navigate sequentially on focus, so that I can tab through web content in a meaningful and efficient sequence.

WCAG 2.4.4: As a user that requires a screen reader, I want a clear anchor link text, so I know what the linked page or button is about. 

Clarify links and link purpose.

## Motivation and Context
<!-- ClickUp ticket IDs are autolinked. So, you only need to list the ticket ID(s) related to this PR -->
[BCHDCC-41](https://app.clickup.com/t/9006094761/BCHDCC-41) - Accessibility Criterion 2.4.4: Link Purpose (In Context)
[BCHDCC-40](https://app.clickup.com/t/9006094761/BCHDCC-40) - Accessibility Criterion 2.4.3: Focus Order

<!--- Why is this change required? What problem does it solve? -->
<!--- Link to any ancillary pieces that you had to use to close the ticket and
      think would be interesting or valuable for readers of this PR. This could
      include things like Airbrake exceptions, ruby/rails documentation, blog
      posts, etc.-->
<!--- Describe any difficulties that arose during creation and any tickets that
      were created as a result. If you have thoughts on how to make those
      difficulties less difficult in the future but not as part of this PR,
      write down those thoughts. -->

## Screenshots
<!--- If this is a UI change, put a relevant screenshot(s) here with a 
      description of what is in it. -->

**Libraries Added**

## Migrations to Run: Yes / **No**

## Tests to Run


## Internal Checklist
<!--- Go over all the following points and make sure you feel comfortable you covered them. -->
- [x] My code follows the code style of this project (https://rubystyle.guide/).
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

emmabjj committed:
Add underlining on hover to make link clear

emmabjj committed:
Merge branch 'develop' into BCHDCC-41-2.4.4-link-purpose

anna-dorigo commented:
I did a quick scan of the changes and I noticed the following:
The main header doesn't indicate that is a link and is clickable, the organization title does.
<img width="664" alt="image" src="https://github.com/user-attachments/assets/80a98354-e471-4111-a0b3-67902ead03a1">

There is a small formatting issue here:
<img width="277" alt="image" src="https://github.com/user-attachments/assets/794e0709-f36b-4af4-b804-9fc312aaab07">

emmabjj committed:
Add spacing to Sign Up instructions

emmabjj committed:
Add more defined role to h1

emmabjj committed:
Merge branch 'BCHDCC-41-2.4.4-link-purpose' of github.com:bchd/bahc-ohana-api into BCHDCC-41-2.4.4-link-purpose

emmabjj commented:
@anna-dorigo - I added those fixes 👍 I used another screen reader to test the h1 link screen reading and it read the h1 correctly, I added the role of link to that h1 just to further clarify its role. 
I think the h1 not being read as a link has to do with that particular screen reader, particularly because there's been discussion in the accessibility community about making h1s links, though I think this use-case is fine.

emmabjj committed:
Merge branch 'develop' into BCHDCC-41-2.4.4-link-purpose

anna-dorigo reviewed (approved):
👍 Great work Emma! one more down!
